avocado/core/task/statemachine.py

```diff
@@ -6,8 +6,9 @@
 class TaskStateMachine:
     """Represents all phases that a task can go through its life."""
 
-    def __init__(self, tasks):
+    def __init__(self, tasks, status_repo):
```

Comment:
Nitpick: About this and the `examples/nrunner/task-state-machine.py`: could you please explain your comment?

> If you run that python script, you will see it already reflects what the state machine does today.

IIUC, this script is not using any of the core code (and your code), right? Is just a standalone script doing a "conceptual" state-machine. So,  I find strange having old examples in our repository. I would a) update this example with proper code or b) remove completely this example. 

Not sure if I understood why this example is still there.

Reply:
Thanks for https://github.com/avocado-framework/avocado/pull/4577!